website/docs/r/lb_listener.html.markdown

```diff
@@ -185,10 +185,27 @@ The following arguments are supported:
 Action Blocks (for `default_action`) support the following:
 
 * `type` - (Required) The type of routing action. Valid values are `forward`, `redirect`, `fixed-response`, `authenticate-cognito` and `authenticate-oidc`.
-* `target_group_arn` - (Optional) The ARN of the Target Group to which to route traffic. Required if `type` is `forward`.
+* `target_group_arn` - (Optional) The ARN of the Target Group to which to route traffic. Specify only if `type` is `forward` and you want to route to a single target group. To route to one or more target groups, use a `forward` block instead.
+* `forward` - (Optional) Information for creating an action that distributes requests among one or more target groups. Specify only if `type` is `forward`. If you specify both `forward` block and `target_group_arn` attribute, you can specify only one target group using `forward` and it must be the same target group specified in `target_group_arn`.
```

Comment:
Specifying both `target_group_arn` and `forward` might be confusing to users. Would it be better to allow one or the other?

Reply:
Mmmm... this is how this resource is implemented (see [AWS CLI params](https://docs.aws.amazon.com/cli/latest/reference/elbv2/create-rule.html)). Maybe it would make sense to deprecate `target_group_arn` and move to a `forward` block-only schema, but this would require modifying the API output because if you specify only a `forward` block, the API output will return the `target_group_arn` attribute instead of the specified `forward` block.

Reply:
Ah, interesting. The API documentation I saw didn't mention that, but it's in the CLI documentation.